### PR TITLE
Add documentation for resources and data sources of Cloud Assembly co…

### PR DESCRIPTION
**Pull request by wilsonandvmware**
…nstructs

Added documentations for the following data source and resources.

data "vra_image_profile"
data "vra_network_profile"
data "vra_project"
data "vra_storage_profile"
data "vra_storage_profile_aws"
data "vra_storage_profile_azure"
data "vra_storage_profile_vsphere"
data "vra_zone

resource "vra_flavor_profile"
resource "vra_image_profile"
resource "vra_network_profile"
resource "vra_project"
resource "vra_storage_profile"
resource "vra_storage_profile_aws"
resource "vra_storage_profile_azure"
resource "vra_storage_profile_vsphere"
resource "vra_zone"

Signed-off-by: Xiaosheng Wu <wxiaosheng@vmware.com>

### CODE REVIEW

File: website/docs/r/vra_storage_profile.html.markdown [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+* `name` - A human-friendly name used as an identifier in APIs that support this option.  Only one of 'filter', 'id', 'name' or 'region_id' must be specified.
+
+* `org_id` - The id of the organization this entity belongs to.
+
+* `owner` - Email of the user that owns the entity.
+
+* `tags` - A set of tag keys and optional values that were set on this Network Profile.
+           example:[ { "key" : "ownedBy", "value": "Rainpole" } ]
+
+* `updated_at` - Date when the entity was last updated. The date is ISO 8601 and UTC.
```
Comment (dmettem) [inline website/docs/r/vra_storage_profile.html.markdown:75]: Not all arguments are documented and also some of the required arguments such as name are documented under computed attributes.

File: website/docs/r/vra_storage_profile_aws.html.markdown [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+# Resource: vra_storage_profile_aws
+## Example Usages
+This is an example of how to create a storage profile aws resource.
+
+**Vra storage profile aws:**
+
+```hcl
+# AWS storage profile using generic vra_storage_profile resource. Use 'vra_storage_profile_aws' resource as an alternative.
+resource "vra_storage_profile" "this" {
```
Comment (dmettem) [inline website/docs/r/vra_storage_profile_aws.html.markdown:17]: The example is for creating AWS storage profile using generic `vra_storage_profile`resource. This example along with Azure one using generic `vra_storage_profile` resource should be included in the storage profile resource page.

Replace this example to use `vra_storage_profile_aws` resource and include for both instance store and EBS.

File: website/docs/r/vra_storage_profile_aws.html.markdown [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+* `external_region_id` - The id of the region as seen in the cloud provider for which this profile is defined.
+
+* `links` - HATEOAS of the entity
+
+* `organization_id` - The id of the organization this entity belongs to.
+
+* `owner` - Email of the user that owns the entity.
+
+* `updated_at` - Date when the entity was last updated. The date is ISO 8601 and UTC.
```
Comment (dmettem) [inline website/docs/r/vra_storage_profile_aws.html.markdown:70]: Same comment as before that not all the properties are documented.

File: website/docs/r/vra_storage_profile_azure.html.markdown [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    azureDataDiskCaching = "None"         // Supported Values: None, ReadOnly, ReadWrite
+    azureManagedDiskType = "Standard_LRS" // Supported Values: Standard_LRS, StandardSSD_LRS, Premium_LRS
+    azureOsDiskCaching   = "None"         // Supported Values: None, ReadOnly, ReadWrite
+  }
+
+  tags {
+    key   = "foo"
+    value = "bar"
+  }
+}
```
Comment (dmettem) [inline website/docs/r/vra_storage_profile_azure.html.markdown:34]: Same comment as before to move this to storage profile page and replace it with `vra_storage_profile_azure`.

File: website/docs/r/vra_storage_profile_azure.html.markdown [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+```
+
+A storage profile azure resource supports the following arguments:
+
+## Required arguments
+
+* `name` - A human-friendly name used as an identifier in APIs that support this option.  Only one of 'filter', 'id', 'name' or 'region_id' must be specified.
```
Comment (dmettem) [inline website/docs/r/vra_storage_profile_azure.html.markdown:41]: Remove 'Only one of 'filter', 'id', 'name' or 'region_id' must be specified.' from all the resource documentation files for `name` argument.

File: website/docs/r/vra_zone.html.markdown [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+## Optional arguments
+
+* `description` - A human-friendly description.
+
+* `folder` - The folder relative path to the datacenter where resources are deployed to. (only applicable for vSphere cloud zones)
+
+* `placement_policy` - The id of the region for which this zone is defined
+
+* `region_id` - A link to the region that is associated with the storage profile.                      example:[ { "key" : "ownedBy", "value": "Rainpole" } ]
```
Comment (dmettem) [inline website/docs/r/vra_zone.html.markdown:48]: Should be a required argument.

**Review by dmettem — COMMENTED** [review]
Most of these comments are duplicates. For each of the comments, fix in all the applicable files.

File: website/docs/d/vra_image_profile.html.markdown [hunk 7]
```diff
@@ -0,0 +1,53 @@
+---
+layout: "vra"
+page_title: "VMware vRealize Automation: vra_image_profile"
+sidebar_current: "docs-vra-datasource-vra-image_profile"
```
[the remark this answers is not in the document]
Comment (wilsonandvmware) [reply website/docs/d/vra_image_profile.html.markdown:4]: Done.

File: website/docs/d/block_device.html.markdown [hunk 8]
```diff
@@ -44,7 +44,7 @@ data "vra_block_device" "this" {
 
 This is an example of how to create a block device resource and filter it as a data source using its name.
 NOTE: The block device resource need not be created through terraform.
-To create a block device, follow the resource block device documentation
+To create a block device, follow the resource block device documentation.
```
[the remark this answers is not in the document]
Comment (wilsonandvmware) [reply website/docs/d/block_device.html.markdown:47]: I modified block_device doc in the other PR, here I'm only adding a period sign.

File: website/docs/r/vra_storage_profile_azure.html.markdown (hunk 5 above)
Comment (wilsonandvmware) [reply website/docs/r/vra_storage_profile_azure.html.markdown:41]: i also removed it for `region-id` or `id` argument

**Review by dmettem — APPROVED** [review]
LGTM.

**Pull request merged**